datashader/resampling.py

```diff
@@ -990,3 +990,71 @@ def infer_interval_breaks(coord, axis=0):
     trim_last = tuple(slice(None, -1) if n == axis else slice(None)
                       for n in range(coord.ndim))
     return np.concatenate([first, coord[trim_last] + deltas, last], axis=axis)
+
+
+@ngjit
+def infer_interval_breaks_2d(coord):
+    """
+    Optimized Numba version for 2D arrays.
+    Equivalent to applying infer_interval_breaks sequentially on axis=1 then axis=0.
+    Combines both passes for interior points to avoid intermediate array.
+
+    Parameters:
+    -----------
+    coord : ndarray
+        2D coordinate array representing grid centers
+
+    Returns:
+    --------
+    ndarray
+        2D array with interval boundaries, shape (m+1, n+1)
+    """
+    m, n = coord.shape
+
+    assert m > 1
```

Comment:
afaict quadmesh expects at least size-2 dimensions even though no clear error message is raised saying this